lock/lock.go

```diff
@@ -83,23 +83,31 @@ func (c *Checker) CheckTableLock(db, table string, privilege mysql.PrivilegeType
 
 	if privilege == mysql.SelectPriv {
 		switch tb.Meta().Lock.Tp {
-		case model.TableLockRead, model.TableLockWriteLocal:
+		case model.TableLockRead, model.TableLockWriteLocal, model.TableLockReadOnly:
 			return nil
 		}
 	}
+	if readOnlyLock && tb.Meta().Lock.Tp == model.TableLockReadOnly {
+		return nil
+	}
+
 	return infoschema.ErrTableLocked.GenWithStackByArgs(tb.Meta().Name.L, tb.Meta().Lock.Tp, tb.Meta().Lock.Sessions[0])
 }
 
-func checkLockTpMeetPrivilege(tp model.TableLockType, privilege mysql.PrivilegeType) bool {
+func checkLockTpMeetPrivilege(tp model.TableLockType, privilege mysql.PrivilegeType, readOnlyLock bool) bool {
```

Comment:
Currently, when calling this function, `readOnlyLock` will always be false, does this expected?

Reply:
Yes, I think this modification is not necessary.